dask/dataframe/groupby.py

```diff
@@ -1950,7 +1950,9 @@ def unique(self, split_every=None, split_out=1):
 
 
 def _unique_aggregate(series_gb, name=None):
-    ret = pd.Series({k: v.explode().unique() for k, v in series_gb}, name=name)
+    ret = series_gb.obj._constructor(
```

Comment:
Wonder if we should just grab the `type` instead of calling a private method from another library. WDYT?

```suggestion
    ret = type(series_gb.obj)(
```

Reply:
This is a good point @jakirkham

FWIW we use `_constructor` other places in `dask.dataframe`, for example:

https://github.com/dask/dask/blob/d6bbbb08c92652eae2820e93edc2f3fe502391d3/dask/dataframe/accessor.py#L131

and implement `_constructor` ourselves

https://github.com/dask/dask/blob/965e35fe5a9423f550991bde08116a4984498e68/dask/dataframe/core.py#L345-L347

@TomAugspurger do you have a sense for how safe methods like `_constructor`, `_constructor_sliced`, etc. are in pandas?

Reply:
To be clear, I don't mean for this question to block merging in this PR

Reply:
They're documented as part of the public API: https://pandas.pydata.org/docs/development/extending.html?highlight=_constructor#override-constructor-properties, so it's fine to use. Only prefixed with an underscore since most users will never use or care about them.

Reply:
It's totally fine to use `type` instead of `_constructor`, but we *must* use `_constructor_sliced` in some places, and have therefore confirmed with Tom/pandas-devs that these attributes can be treated as "public."

Reply:
Ah ok. Didn't not realize we were allowed to use this. The `_` made me think it was private. Ok with whatever people here decide